src/resolve.ts

```diff
@@ -0,0 +1,47 @@
+import {NonspatialScaleChannel, SCALE_CHANNELS, SPATIAL_SCALE_CHANNELS, SpatialScaleChannel, UNIT_SCALE_CHANNELS} from './channel';
+import * as log from './log';
+import {contains} from './util';
+
+export type ResolveMode = 'independent' | 'shared';
+
+export interface SpatialResolve {
+  scale: ResolveMode;
```

Comment:
Why is scale required?    

Suppose a scale is shared and I want to override its axes to be independent, I don't need to specify scale resolve.